docs/mkdocs/docs/technical/contributing.md

```diff
@@ -0,0 +1,195 @@
+* [CLion](https://www.jetbrains.com/help/clion/cmake-presets.html)
+
+And it's equally easy to use on the command line ([see example](#compiling-step-by-step)).
+
+We already ship a `CMakePresets.json` in the cpp directory, which is used by our builds.
+You can add a `CMakeUserPresets.json` in the same directory for local overrides.
+Inheritance is supported.
+
+Example `CMakeUserPresets.json`:
+
+<details>
+<summary>Windows Preset to specify a Python version</summary>
+
+```json
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "alt-vcpkg-debug:py3.10",
+      "inherits": "windows-cl-debug",
+      "cacheVariables": {
+        "Python_ROOT_DIR": "C:\\Program Files\\Python310"
+      },
+      "environment": {
+        "PATH": "C:\\Users\\me\\AppData\\Roaming\\Python\\Python310\\Scripts;C:\\Program Files\\Python310;$penv{PATH}",
+        "PYTHONPATH": "C:\\Program Files\\Python310\\Lib;C:\\Users\\me\\AppData\\Roaming\\Python\\Python310\\site-packages"
+      }
+    }
+  ],
+  "buildPresets": [
+    {
+      "name": "alt-vcpkg-debug:py3.10",
+      "configurePreset": "alt-vcpkg-debug:py3.10",
+      "inheritConfigureEnvironment": true
+    }
+  ]
+}
+```
+</details>
+
+## vcpkg caching
+We use [vcpkg](https://vcpkg.io/) to manage the C++ dependencies.
+
+The CMakePresets we provided use the `cpp/vcpkg` directory as the toolchain, so
+it's not necessary to separately download and install vcpkg.
+(You can, however, maintain a separate vcpkg install and override the CMake presets.)
+
+This setup downloads the dependencies' source packages from the Internet and compiles them locally during CMake
+configuration, so it's critical to cache this process to save time (and to comply with any local IT policy).
+
+Please refer to below documentation:
+* [Binary caching](https://learn.microsoft.com/en-us/vcpkg/users/binarycaching)
+* [Asset caching](https://learn.microsoft.com/en-us/vcpkg/users/assetcaching)
+
+For example, you can use a network share as the caching folder and put the caching configuration environment
+variables into the CMake presets.
+
+Please also note compiling the dependencies uses a lot of disk space.
+Once CMake configuration is done and all the dependencies saved into the binary cache,
+you can remove the `cpp\vcpkg\buildtrees` folder.
+
+## pybind11 configuration
+We augmented pybind11's [Python discovery](https://pybind11.readthedocs.io/en/stable/compiling.html#building-with-cmake)
+with our own [PythonUtils](https://github.com/man-group/ArcticDB/blob/master/cpp/CMake/PythonUtils.cmake) to improve
+diagnostics.
+Please pay attention to warning messages from `PythonUtils.cmake` in the CMake output which highlights any configuration
+issues with Python.
+
+We recommend:
+
+* Using the latest [manylinux](#manylinux-docker-image) image for Linux development where possible;
+* Or, getting the latest CMake (at least 3.18) and make sure it's the first on your PATH
```

Comment:
don't we require 3.21+?

Reply:
3.21 is to support version 3 of the CMakePresets.
3.18 is sufficient to find the `Development.Module`.